connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/MySQLIntegrationSuite.scala

```diff
@@ -241,6 +241,58 @@ class MySQLIntegrationSuite extends DockerJDBCIntegrationV2Suite with V2JDBCTest
     assert(rows10(0).getString(0) === "amy")
     assert(rows10(1).getString(0) === "alex")
   }
+
+  test("SPARK-50793: MySQL JDBC Connector failed to cast some types") {
+    val tableName = catalogName + ".test_cast_function"
+    withTable(tableName) {
+      val stringValue = "0"
+      val stringLiteral = "'0'"
+      val longValue = 0L
+      val binaryValue = Array[Byte](0x30)
+      val binaryLiteral = "x'30'"
+      val doubleValue = 0.0
+      val doubleLiteral = "0.0"
+      // CREATE table to use types defined in Spark SQL
+      sql(s"""CREATE TABLE $tableName (
```

Comment:
nit: let's avoid this multiline string

Reply:
The template is longer than 100 limit and fails the style check. Unless we change column name to something like c1,c2,c3,c4. What do you think about changing to these column names?

Reply:
From the next comment I kind of get a idea that you might mean using string concatenation instead of multiline string. Since it works anyway, let me make the changes based on this assumption. In the case of wrong, I would be happy to change these accordingly.

Reply:
I pushed a commit to concatenate two strings, PTAL